src/target/stm32f4.c

```diff
@@ -593,14 +587,17 @@ static bool stm32f4_option_write(target *t, uint32_t *val, int count)
 	int spinindex = 0;
 	tc_printf(t, "Erasing flash... This may take a few seconds.  ");
 	/* Read FLASH_SR to poll for BSY bit */
-	while(target_mem_read32(t, FLASH_SR) & FLASH_SR_BSY) {
-		platform_delay(100);
-		tc_printf(t, "\b%c", spinner[spinindex++ % 4]);
-		if(target_check_error(t)) {
+	uint32_t sr;
+	do {
+		sr = target_mem_read32(t, FLASH_SR);
+		if ((sr & SR_ERROR_MASK) || target_check_error(t)) {
 			tc_printf(t, " failed\n");
 			return false;
 		}
-	}
+		tc_printf(t, "\b%c", spinner[spinindex++ % 4]);
```

Comment:
With this restructure, this can be replaced with `target_print_progress` which would eliminate another of these manual spinner implementations.